config/routes.rb

```diff
@@ -1,7 +1,21 @@
 Rails.application.routes.draw do
-  resources :users
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  # resources :category_records
+  # resources :records
+  # resources :categories
 
-  # Defines the root path route ("/")
-  # root "articles#index"
-end
+  devise_for :users
+  devise_scope :user do
+    authenticated :user do
+      root 'categories#index', as: :authenticated_root
+    end
+    unauthenticated do
+      root 'splashes#index', as: :unauthenticated_root
+    end
+    get '/users/sign_out' => 'devise/sessions#destroy'
+  resources :users, only: [:index]
+  end
+  resources :categories do 
+    # resources :records, except: [:index]
+    resources :records
+  end
```

Comment:
- [ ] @Kelvin-Ben, we really need to make sure that only necessary routes are used on this project. By default, Rails creates routes for all seven of the default actions (index, show, new, create, edit, update, and destroy) for every RESTful route in your application. You can use the :only and :except options to fine-tune this behavior. The :only option specifies that only certain routes should be generated. This can really save our memory space

Reply:
@Deepakdanger, you know at first, I did not have full information about the purpose of RESTful routes but you nade it clear for me. Thank you